### PR TITLE
feat: Bookmarks

### PR DESCRIPTION
**Pull request by JuliusKryger**
This pull request introduces a comprehensive set of changes to add and test bookmark management functionality in the CitMovie application. The changes include new controllers, managers, repositories, and unit/integration tests.

### Bookmark Management Functionality:

* [`src/CitMovie.Api/Controller/BookmarkController.cs`](diffhunk://#diff-27db7e5d9e3e2030e79d457e4a6706ff637135f61c780044404aedfa9d83826fR1-R76): Added a new `BookmarkController` to handle CRUD operations for bookmarks with user authorization.
* [`src/CitMovie.Business/Managers/BookmarkManager.cs`](diffhunk://#diff-96b85ee29d2cb7ba6835a24858503ba1299fa0602e5a021cc938e5df79b7d1efR1-R96): Implemented `BookmarkManager` to manage bookmark operations like create, update, get, and delete.
* [`src/CitMovie.Business/Managers/IBookmarkManager.cs`](diffhunk://#diff-33981c4248a2819b16f90b615c7d09b592a65330a62b1d00878a170bb8878abcR1-R45): Defined the `IBookmarkManager` interface to standardize bookmark management operations.
* [`src/CitMovie.Business/CitMovieServiceCollectionExtensions.cs`](diffhunk://#diff-dabfe6c1fef4740e18f2895f06dcf816b3560336c995de957529a0af40a905c1R3-R35): Updated service collection extensions to include bookmark-related services. [[1]](diffhunk://#diff-dabfe6c1fef4740e18f2895f06dcf816b3560336c995de957529a0af40a905c1R3-R35) [[2]](diffhunk://#diff-dabfe6c1fef4740e18f2895f06dcf816b3560336c995de957529a0af40a905c1R45-R49)

### Testing:

* [`src/CitMovie.Api.Tests/Integration/BookmarkControllerIntegrationTests.cs`](diffhunk://#diff-7231e51e59a83c6bc3a6883c266b2583b20094a7f32224e881f07403e4d416c8R1-R54): Added integration tests for `BookmarkController` to validate end-to-end scenarios.
* [`src/CitMovie.Api.Tests/Unit/BookmarkControllerTests.cs`](diffhunk://#diff-b3de429d773c00648c5f5e21ce2d513a68ac3ade59889f375d003d4dd8fee157R1-R171): Added unit tests for `BookmarkController` to validate individual functionalities and edge cases.
* [`src/CitMovie.Api.Tests/CitMovie.Api.Tests.csproj`](diffhunk://#diff-74df60f9e4ce5141a3fc45794f671188c26d9facd0ff8a34470be5fb889a325fR13): Updated project references and added `FakeItEasy` for mocking dependencies in tests. [[1]](diffhunk://#diff-74df60f9e4ce5141a3fc45794f671188c26d9facd0ff8a34470be5fb889a325fR13) [[2]](diffhunk://#diff-74df60f9e4ce5141a3fc45794f671188c26d9facd0ff8a34470be5fb889a325fR28-R29)

These changes ensure that bookmark management is fully integrated and tested within the CitMovie application.

### CODE REVIEW

File: src/CitMovie.Business/CitMovieServiceCollectionExtensions.cs
Comment (sofushn) [inline src/CitMovie.Business/CitMovieServiceCollectionExtensions.cs:1]: Why has this been changed?

File: src/CitMovie.Business.Tests/BookmarkManagerTests.cs
Comment (sofushn) [inline src/CitMovie.Business.Tests/BookmarkManagerTests.cs:1]: Everything in this file has been commented out

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace CitMovie.Api.Controllers
+{
+    [ApiController]
+    [Route("api/bookmarks")]
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:12]: Should the bookmarks endpoints not be under `api/users/{userId}/bookmarks`

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public class BookmarkController : ControllerBase
+    {
+        private readonly IBookmarkManager _bookmarkManager;
+
+        public BookmarkController(IBookmarkManager bookmarkManager) =>
+            _bookmarkManager = bookmarkManager;
+
+        private int GetUserId() =>
+            int.Parse(User.FindFirstValue("user_id") ?? throw new UnauthorizedAccessException("User ID not found"));
+
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:23]: Please make use of the user_scope policy to check if user is allowed to use an endpoint, see follow or user controller for example of how to implement

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        [HttpGet]
+        public async Task<IActionResult> GetUserBookmarks()
+        {
+            var userId = GetUserId();
+            var bookmarks = await _bookmarkManager.GetUserBookmarksAsync(userId);
+            return Ok(bookmarks);
+        }
+
+        [HttpPut("{id}")]
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:51]: Updates should be done with the patch http method, see create user flow for example if implementation

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            var updatedBookmark = await _bookmarkManager.UpdateBookmarkAsync(id, note);
+            return updatedBookmark == null ? NotFound() : Ok(updatedBookmark);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteBookmark(int id)
+        {
+            var bookmark = await _bookmarkManager.GetBookmarkAsync(id);
+            if (bookmark == null || bookmark.UserId != GetUserId())
+                return Unauthorized();
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:70]: Should be not found if bookmark don't exist

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
+using CitMovie.Business.Managers;
+using CitMovie.Models.DTOs;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace CitMovie.Api.Controllers
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:9]: Namespace should not include controllers

**Review by sofushn — CHANGES_REQUESTED** [review]
Love that you have included integration tests and xml documentation for the interfaces.

Please look at how other files uses global usings to reduce using imports in each file

**Review by sofushn — APPROVED** [review]
